ZAPD/ZTexture.cpp

```diff
@@ -756,17 +797,39 @@ Declaration* ZTexture::DeclareVar(const std::string& prefix,
 std::string ZTexture::GetBodySourceCode() const
 {
 	std::string sourceOutput;
+	std::string typeSizeStr =
+		Globals::Instance->outputPath.stem().stem().stem().extension().c_str();
+	TextureTypeSize texTypeSize = TextureTypeSize::Size64;
 
-	for (size_t i = 0; i < textureDataRaw.size(); i += 8)
+	if (!typeSizeStr.empty())
 	{
-		if (i % 32 == 0)
-			sourceOutput += "    ";
-
-		sourceOutput +=
-			StringHelper::Sprintf("0x%016llX, ", BitConverter::ToUInt64BE(textureDataRaw, i));
+		texTypeSize = static_cast<TextureTypeSize>(StringHelper::StrToL(typeSizeStr.substr(2)));
+	}
+	for (size_t i = 0; i < textureDataRaw.size();
```

Comment:
This assumes that the texture fits into a whole number of elements of the type it's been given. Currently it'll be UB otherwise, at least. But this should be tested both when extracting the texture and when building it.